controllers/reconcilers/configuration/configuration_reconciler.go

```diff
@@ -609,8 +609,8 @@ func (r *Reconciler) deleteUnrequestedCredentialSecrets(ctx context.Context, cr
 }
 
 func (r *Reconciler) getResourcesUrl(repo *v1.RepositoryInfo, cr *v1.Observability) (*url.URL, error) {
-	// assume container resources if there is no access token
-	if repo.AccessToken == "" {
+	// assume container resources if repo is not GitHub
```

Comment:
the logic could be further improved by having a configuration that indicates the type of the repo:
- `obsevability_resourcs_repo_type: git|container_image|file| ....` 

The type config could be passed in some secret and used here to know where & how to fetch the observability resources

Reply:
this way the git repo is not only limited to github for example and we are explicit about this configuration.

Reply:
Thanks @machi1990 that would be a good improvment to make